caas/kubernetes/provider/application/application.go

```diff
@@ -1089,6 +1090,40 @@ func (a *app) Service() (*caas.Service, error) {
 	}, nil
 }
 
+func (a *app) computeStatus(ctx context.Context, client kubernetes.Interface, now time.Time) (string, status.Status, time.Time, error) {
+	jujuStatus := status.Waiting
+	switch a.deploymentType {
+	case caas.DeploymentStateful:
+		ss, err := a.getStatefulSet()
+		if err != nil {
+			return "", jujuStatus, now, errors.Trace(err)
+		}
+		if ss.GetDeletionTimestamp() != nil {
+			jujuStatus = status.Terminated
+		} else if ss.Spec.Replicas != nil && ss.Status.ReadyReplicas == *ss.Spec.Replicas {
+			jujuStatus = status.Active
```

Comment:
an application can be active even if all pods are not yet ready
i think we'd want "allocating" if ReadyReplicas is 0 and "active" if there's at least 1
if something different is needed, then it's up to the charm to set application status itself